tony-core/src/main/java/com/linkedin/tony/events/EventHandler.java

```diff
@@ -86,25 +93,33 @@ public void run() {
 
     try {
       dataFileWriter.close();
-      out.close();
+      if (out != null) {
+        out.close();
+      }
     } catch (IOException e) {
       LOG.error("Failed to close writer", e);
     }
 
-    // At this point, historyFile should be set
+    // If setupThread method fails to create tmpHistFile,
+    // return immediately since we don't have any file to begin with
+    if (tmpHistFile == null) {
+      return;
+    }
+
+    // At this point, finalHistFile should be set
     // If not, then discard all events
-    if (historyFile == null) {
+    if (finalHistFile == null) {
```

Comment:
Even if finalHistFile is not set, I don't think we should just delete the written events. Just leave them as is and they can get cleaned up during retention (#98). In fact, the written events should help us debug why the finalHistFile wasn't set (seeing which events have been written can help us pinpoint when in the application execution it crashed).